### PR TITLE
Finish advanced login

### PR DESCRIPTION
**Pull request by NuongNguyen1996**
ready

### CODE REVIEW

**Comment by NuongNguyen1996** [comment]
ready

**Comment by nvtanh** [comment]
commented
conflict + 50 files
tách pull ra e, tối đa 15 files

**Comment by NuongNguyen1996** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -1,15 +1,42 @@
 class User < ApplicationRecord
-  before_save{email.downcase!}
+ attr_accessor :remember_token
```
Comment (dothidiemthao) [inline app/models/user.rb]: sai indent

File: app/views/shared/_error_messages.html.erb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <% if @user.errors.any? %>
   <div id="error_explanation">
     <div class="alert alert-danger">
-      The form contains <%= pluralize(@user.errors.size, "error") %>.
+      The form contains <%= pluralize(@user.errors.count, "error") %>.
```
Comment (dothidiemthao) [inline app/views/shared/_error_messages.html.erb]: i18n, dùng size nhé

**Comment by dothidiemthao** [comment]
commented

**Comment by NuongNguyen1996** [comment]
ready

**Review by dothidiemthao — APPROVED** [review]

**Comment by dothidiemthao** [comment]
merged

**Pull request merged**